src/ripple/app/paths/cursor/RippleLiquidity.cpp

```diff
@@ -209,32 +204,33 @@ rippleQuality (
     SField const& sfLow,
     SField const& sfHigh)
 {
-    std::uint32_t uQuality (QUALITY_ONE);
+    if (destination == source)
+        return parityRate;
```

Comment:
As long as you did this simplification, we may as well re-write the rest of the function to return early when we can. Cherry pick this if you want: https://github.com/seelabs/rippled/commit/0a42d3c51085421cd1d3a39306edb3cb29ca8b0a (fine as-is though).